evil-common.el

```diff
@@ -867,9 +867,35 @@ Inhibits echo area messages, mode line updates and cursor changes."
   `(let ((evil-no-display t))
      ,@body))
 
-(defun evil-num-visible-lines ()
-  "Returns the number of currently visible lines."
-  (- (window-height) 1))
+(defvar evil-cached-header-line-height nil
+  "Cached height of the header line.")
+
+(defun evil-header-line-height ()
```

Comment:
@bet4it It looks like this function doesn't give a reliable result, leading to the bug that this PR is supposed to fix.

To elaborate, whenever I enable [centaur-tabs](https://github.com/ema2159/centaur-tabs), which uses the header line to show tabs, I still encounter the bug on `evil-scroll-up` and `evil-scroll-down`. The header line by `centaur-tabs`  looks really tall compared to the plain header line, and yet `evil-header-line-height` always returns the same value.

<img width="469" alt="Screen Shot 2019-12-31 at 09 21 05" src="https://user-images.githubusercontent.com/9099577/71628921-74f97c80-2baf-11ea-9dec-0297e316b6d0.png">

Figure 1: the header line by `centaur-tabs`, along with the value of `evil-header-line-height`, `window-height`., and `window-body-height`.

<img width="469" alt="Screen Shot 2019-12-31 at 09 20 39" src="https://user-images.githubusercontent.com/9099577/71628926-7b87f400-2baf-11ea-98c4-6c4eb08f5631.png">

Figure 2: the regular header line, along with the value of `evil-header-line-height`, `window-height`., and `window-body-height`.

Reply:
If I use `window-header-line-height` instead, the bug will be properly fixed. Will submit a PR.